### ISSUE TITLE
Azure Data Studio cant install updates

### ISSUE THREAD

**Issue by DeveloperLookBook** [microsoft/azuredatastudio#25837]
Type: <b>Bug</b>

When I install updates for Azure Data Studio, it says that everything was installed but in a few moment later says that I have updates again. This repeats infinitely after each update. 

Azure Data Studio version: azuredatastudio 1.45.1 (88c21b1725a3e79440027bdb7b5a55fb036be0e2, 2023-08-03T00:42:56.773Z)
OS version: Darwin arm64 23.5.0
Restricted Mode: No
Preview Features: Enabled
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Max (10 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|2, 2, 3|
|Memory (System)|64.00GB (12.40GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
copilot|Git|1.128.504

</details>

**Comment by kburtram** [comment]
@caohai do you have any idea what's going on here?

**Comment by caohai** [comment]
Hi @DeveloperLookBook , after installing the update, are you seeing the same `1.45.1` version? If so the local installation might have some issue applying the update.

I just tested by installing `1.45.1` on my MacBook (intel) and was able to successfully update to the latest version.

I'd suggest you to manually install the latest `1.49.1` version as `1.45.1` is more than 1 year old.
https://learn.microsoft.com/en-us/azure-data-studio/download-azure-data-studio

@kburtram I don't think this is an issue with the update infra as I was not able to repro. The update server response also looks correct to me:
```
GET https://azuredatastudio-update.azurewebsites.net/api/update/darwin-arm64/stable/88c21b1725a3e79440027bdb7b5a55fb036be0e2 HTTP/1.1
```
<img width="930" alt="image" src="https://github.com/user-attachments/assets/8b7db270-0d92-4aeb-8ee2-e4bdcc82a460">

**Comment by Yevhen-Myroshnychenko** [comment]
Hi, @caohai 

Here is how it looks from my side after the updates installation. As you can see Azure Data Studio propose me to install updates again and again, infinitely.

<img width="532" alt="Screenshot 2024-08-16 at 10 29 37" src="https://github.com/user-attachments/assets/98af9dd9-61f3-4b49-ba79-78b2380b22b3">

**Comment by DeveloperLookBook** [comment]
Hi, @caohai

Here is how it looks from my side after the updates installation. As you can see Azure Data Studio propose me to install updates again and again, infinitely.

<img width="532" alt="Screenshot 2024-08-16 at 10 29 37" src="https://github.com/user-attachments/assets/8bf938b9-2f75-4cec-a22d-edf09056ed54">

**Comment by DeveloperLookBook** [comment]
@caohai Also, I don't understand, why do I have a version 1.45.1 that is one year old. I regularly installed updates.  Probably, all this time I was sure that I'm updating it

**Comment by DeveloperLookBook** [comment]
@caohai  There was the same situation with VS Code. Exactly the same!

**Comment by caohai** [comment]
> Hi, @caohai
> 
> Here is how it looks from my side after the updates installation. As you can see Azure Data Studio propose me to install updates again and again, infinitely.
> 
> <img alt="Screenshot 2024-08-16 at 10 29 37" width="532" src="https://private-user-images.githubusercontent.com/41686395/358506153-8bf938b9-2f75-4cec-a22d-edf09056ed54.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM4MjkzMDksIm5iZiI6MTcyMzgyOTAwOSwicGF0aCI6Ii80MTY4NjM5NS8zNTg1MDYxNTMtOGJmOTM4YjktMmY3NS00Y2VjLWEyMmQtZWRmMDkwNTZlZDU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE2VDE3MjMyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTljMDU0OWQxOGZkMTBiNjY5MjNlMWI1MjA4YjhiMzZmZWVhOTU3ZWE3ODMyMWZmODU1NDYzN2U0YmM2YzViMTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-47l44JFBJEo3fiC908BYdfwG2spxfbOPG2MY00LfC8">

Yep this means the update somehow didn't succeed so after installation you are still seeing the same update notification. I'm not sure what has caused this. I'd recommend you manually install the latest version. Can you try it and share if that works for you? https://learn.microsoft.com/en-us/azure-data-studio/download-azure-data-studio

You won't be able to verify if auto-update works with that version given it's the latest but you can always reach back to us later if you see the same behavior when a new version is released in the future.
